presto-main/src/main/java/com/facebook/presto/execution/SqlTaskExecution.java

```diff
@@ -263,6 +250,31 @@ private SqlTaskExecution(
         }
     }
 
+    // this is a separate method to ensure that the `this` reference is not leaked during construction
```

Comment:
// static method to ensure that the `this` reference is not leaked during construction